src/main/java/de/hpi/bpt/argos/parsing/EventTypeParserImpl.java

```diff
@@ -0,0 +1,221 @@
+		}
+	}
+
+	/**
+	 * This method loads the content of a specific file and parses this into an event type.
+	 * @param eventTypeFile - the file to load
+	 */
+	private void loadDefaultEventType(File eventTypeFile) {
+		try {
+			if (!wasModified(eventTypeFile)) {
+				logger.info(String.format("file '%1$s' was loaded already and therefore skipped", eventTypeFile.getName()));
+				return;
+			} else {
+				modify(eventTypeFile);
+			}
+
+			String fileContent = new String(Files.readAllBytes(Paths.get(eventTypeFile.toURI())), StandardCharsets.UTF_8);
+
+			JsonObject jsonEventType = jsonParser.parse(fileContent).getAsJsonObject();
+
+			if (jsonEventType.get(JSON_NAME).getAsString().isEmpty()) {
+				return;
+			}
+
+			List<EventType> eventTypes = PersistenceAdapterImpl.getInstance().getEventTypes();
+
+			for (EventType eventType : eventTypes) {
+				if (eventType.getName().equalsIgnoreCase(jsonEventType.get(JSON_NAME).getAsString())) {
+					return;
+				}
+			}
+
+			createEventType(jsonEventType);
+
+		} catch (Exception e) {
+			LoggerUtilImpl.getInstance().error(logger, String.format("cannot load event type from file '%1$s'", eventTypeFile.getName()), e);
+		}
+	}
+
+	/**
+	 * This method checks whether a file was modified since the last time it was loaded.
+	 * @param file - the file to check for modification
+	 * @return - true, if the file was modified and therefore should be loaded again
+	 */
+	private boolean wasModified(File file) {
+		DataFile dataFile = PersistenceAdapterImpl.getInstance().getDataFile(file.getAbsolutePath());
+
+		return dataFile == null || dataFile.getModificationTimestamp() != file.lastModified();
+	}
+
+	/**
+	 * This method updated the modificationTimestamp for a given file.
+	 * @param file - the file to be updated
+	 */
+	private void modify(File file) {
+		DataFile dataFile = PersistenceAdapterImpl.getInstance().getDataFile(file.getAbsolutePath());
+
+		if (dataFile == null) {
+			dataFile = new DataFileImpl();
+			dataFile.setPath(file.getAbsolutePath());
+		}
+
+		dataFile.setModificationTimestamp(file.lastModified());
+
+		PersistenceAdapterImpl.getInstance().saveArtifacts(dataFile);
+	}
+
+	/**
+	 * This method create a new eventType from it's Json representation.
+	 * @param jsonEventType - the Json representation of the new eventType
+	 */
+	private void createEventType(JsonObject jsonEventType) {
+
+		if (jsonEventType.get(JSON_NAME).getAsString().isEmpty()
+				|| jsonEventType.get(JSON_TIMESTAMP).getAsString().isEmpty()
+				|| jsonEventType.get(JSON_ATTRIBUTES).getAsJsonObject().entrySet().isEmpty()) {
+			return;
+		}
+
+		EventType newEventType = new EventTypeImpl();
+		newEventType.setDeletable(false);
+		newEventType.setShouldBeRegistered(true);
+		newEventType.setName(jsonEventType.get(JSON_NAME).getAsString());
+
+		PersistenceAdapterImpl.getInstance().saveArtifacts(newEventType);
+
+		List<String> usedNames = new ArrayList<>();
```

Comment:
Why do you declare this before saving artifacts to the database? Just put it in front of the for loop please.